### ISSUE TITLE
plotHeatmap throws error with clusterLegend argument with ClusterExperiment object as input

### ISSUE THREAD

**Issue by InfH** [epurdom/clusterExperiment#169]
I am trying to use different color palettes for the colData in a call to `plotHeatmap`.

Using the default color palette works:
`cm3` is a clusterExperiment object

```
plotHeatmap(cm3, clusterSamplesData="dendrogramValue", breaks=.99, sampleData=c("expt", "batch", "chip"), visualizeData=log2(assay(cm3)+1), clusterFeaturesData=ref)
```

When I try to use the `clusterLegend` argument:
`cc` and `col` are color vectors (previously worked with `dualHeatmap` and `aheatmap`)

```
plotHeatmap(cm3, clusterSamplesData="dendrogramValue", breaks=.99, sampleData=c("expt", "batch", "chip"), visualizeData=log2(assay(cm3)+1), clusterFeaturesData=ref, clusterLegend=list(expt=cols, batch=cc,chip=cc))
```

I get the following error:

```
Error in .local(data, ...) : 
  formal argument "clusterLegend" matched by multiple actual arguments
```

**Comment by epurdom** [comment]
The matrix version of plotHeatmap still works like the old dualHeatmap. Currently, plotHeatmap when applied to a clusterExperiment object pulls from the clusterLegend(cm3) to determine the clusterLegend. However, clusterLegend of a clusterExperiment object only stores colors for the clusters, and not for the sampleData (found in colData(cm3). But this is clearly a problem for plotHeatmap when you want to change the colors for sampleData. 

I will mark this as an enhancement and look and see how difficult this is to fix.

**Comment by epurdom** [comment]
Sorry, I spoke too soon. Refreshing my memory on the code, I see that it should allow the user to give a clusterLegend. I will look into this.

**Comment by epurdom** [comment]
I have fixed this bug so now should work (develop 949d17796d).
